### ISSUE TITLE
bug(all): High Contrat Mode is broken?! after page refresh

### ISSUE THREAD

**Issue by Totati** [angular/components#25580]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Enabling High Contrast after page load produces different styles than enabling it before (or after page refresh).
After page load the Button page of docs looks like this:
![image](https://user-images.githubusercontent.com/17689719/188420511-acd0d24d-2f57-4e4e-8a91-f2c5ef174725.png)
All buttons have outline and the currently focused element (Overview tab) has a nice strong focus indicator.

Hitting F5 and they are all gone
![image](https://user-images.githubusercontent.com/17689719/188420914-d384b57f-8004-4a80-9730-bc2e74f8a06a.png)
The currently focused element has a bit more contrast but that's all.

Other components have malfunction too, like FormField

Before refresh:
![image](https://user-images.githubusercontent.com/17689719/188421940-4c6b5314-94f4-408a-a5d4-59d5a19954dd.png)
After refresh:
![image](https://user-images.githubusercontent.com/17689719/188421811-6c758a8f-6bdd-44b8-a7e1-3b8c136f30f8.png)

### Reproduction

Steps to reproduce:
1. Go to https://material.angular.io/components/button/overview
2. Enable High Contrast (shortcut: Alt key + left Shift key + Print screen)
3. Check the styles
4. Press F5

### Expected Behavior

I think they should look like the first time, and the styles after F5 are not accessible.

### Actual Behavior

They differ

### Environment

- Angular: 14.x
- CDK/Material: 14.x
- Browser(s): Microsoft Edge Version 105.0.1343.25 (Official build) (64-bit) / Google Chrome Version 105.0.5195.102 (Official Build) (64-bit)
- Operating System (e.g. Windows, macOS, Ubuntu): 
Edition	Windows 11 Pro
Version	21H2
OS build	22000.856
Experience	Windows Feature Experience Pack 1000.22000.856.0

### Note

This might be caused by the browser or Windows. `HighContrastModeDetector` can't detect it beacuse the coerced color for me is not `rgb(0,0,0)` but `rgb(32,32,32)`. I could not find the reason it works when I dynamically change the contrast, as the debugger won't stop at any of my breakpoints. However when I attach a breakpoint to the `body` itself (like right click on the node and break on attribute modifications) it breaks and works like on page refresh. It stops [HERE](https://github.dev/angular/components/blob/f379e803fd8c3e4298a244f3e438ec249f94cab5/src/cdk/a11y/high-contrast-mode/high-contrast-mode-detector.ts#L111) with the breakpoint on the element, but not when I mark it.

### Note 2

It is caused by Windows. In Windows 11 they have new high contrast themes.
![image](https://user-images.githubusercontent.com/17689719/188444113-86c69043-f577-4182-885c-ef593d718acb.png)
